src/dra/4A538.c

```diff
@@ -1847,69 +1849,85 @@ void RenderPrimitives(void) {
             D_8003C0EC[3]--;
         }
     } else {
-        var_a1 = 0;
-        i = 0;
-        prim = g_PrimBuf + i;
-        for (; i < MAX_PRIM_COUNT; prim++, i++) {
-            primType = prim->type;
-            if (!primType) {
+        var_a2 = 0;
+        var_s1 = 0;
+        for (var_s0 = &g_PrimBuf[0]; var_s1 < MAX_PRIM_COUNT; var_s0++,
```

Comment:
Shall we rename some vars back to the previous names? I think `var_s0` as `prim` or `var_s1` as `i` and so on are a bit clearer.

Reply:
Good call. Done!